### PR TITLE
fix(storage): resolve to fetch instead of xhr in some case of RN

### PR DESCRIPTION
**Pull request by AllanZhengYP**
#### Description of changes

The Storage package resolves to fetch handler instead of XHR in some cases of React Native. This causes the `onProgress` callback not invoked.

#### Description of how you validated changes
Integration test

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by stocaaro — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Review by ashwinkumar6 — APPROVED** [review]
Thank you!

**Review by kvramyasri7 — APPROVED** [review]
Thank you for the Fix @AllanZhengYP  🥳

**Comment by AllanZhengYP** [comment]
Pending test SSR for in Next.js

**Review by HuiSF — APPROVED** [review]

File: packages/storage/src/providers/s3/utils/client/runtime/package.json [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-	},
-	"browser": {
-		"./": "../../../../../../lib-esm/providers/s3/utils/client/runtime/index.browser.js",
-		"fast-xml-parser": false,
-		"buffer": false
-	},
-	"module": "../../../../../../lib-esm/providers/s3/utils/client/runtime/index.browser.js"
+    "name": "@aws-amplify/storage/src/providers/s3/utils/client/runtime",
+    "react-native": {
+        "./index.js": "./index.native.js",
```
Comment (kvramyasri7) [inline packages/storage/src/providers/s3/utils/client/runtime/package.json:4]: More Cleaner 👌

File: packages/storage/src/providers/s3/utils/client/runtime/package.json [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
+        "fast-xml-parser": "fast-xml-parser",
+        "buffer": "buffer"
+    },
+    "browser": {
+        "./index.js": "./index.browser.js",
+        "fast-xml-parser": false,
+        "buffer": false
```
Comment (kvramyasri7) [inline packages/storage/src/providers/s3/utils/client/runtime/package.json:11]: Is these false values for readability as the default value is false ?

**Review by kvramyasri7 — APPROVED** [review]

**Review by ashwinkumar6 — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**